sympy/physics/mechanics/kane.py

```diff
@@ -451,11 +451,13 @@ def get_partial_velocity(body):
             MMi = MM[:p, :]
             MMd = MM[p:o, :]
             MM = MMi + (self._Ars.T * MMd)
+            # Apply the same to nonMM
+            nonMM = nonMM[:p, :] + (self._Ars.T * nonMM[p:o, :])
 
         self._bodylist = bl
         self._frstar = fr_star
         self._k_d = MM
-        self._f_d = -msubs(self._fr + self._frstar, udot_zero)
+        self._f_d = -(self._fr - nonMM)
```

Comment:
One more thought. Are you certain that this manages the auxiliary speeds?

Reply:
Good point, I'll look into it.

Reply:
Good point, I'll look into it.

Reply:
Let's look from an implementation perspective.
```
self._f_d = -msubs(self._fr + self._frstar, udot_zero)
```
 can be rewritten to
```
self._f_d = -msubs(self._fr + dep_transform(-(MM * msubs(Matrix(self._udot), uauxdot_zero) + nonMM)), udot_zero)
```
where [`dep_transform`](https://github.com/sympy/sympy/blob/4f48aad04cdb34e31e821ef9f6e9aa1af0ac637d/sympy/physics/mechanics/kane.py#L449) is adding the columns belonging to the dependent part to the independent part by multiplying with `self._Ars.T`. This equation we can again rewrite to as you may split this transform on additions etc.
```
self._f_d = -msubs(self._fr - dep_transform(MM * msubs(Matrix(self._udot), uauxdot_zero)) - dep_transform(nonMM)), udot_zero)
```
The `msubs` can also be split:
```
self._f_d = -(msubs(self._fr, udot_zero) - msubs(dep_transform(MM * msubs(Matrix(self._udot), uauxdot_zero)), udot_zero) - msubs(dep_transform(nonMM)), udot_zero))
```
So now we have three terms:
1. `msubs(self._fr, udot_zero)`
2. `msubs(dep_transform(MM * msubs(Matrix(self._udot), uauxdot_zero)), udot_zero)`
3. `msubs(dep_transform(nonMM)), udot_zero)`

For term 1 see [the previous discussion](https://github.com/sympy/sympy/pull/24792#issuecomment-1453213005), for why you can write it as just `self._fr`.

For term 2 I'm honestly a bit suprised about this part: `msubs(Matrix(self._udot), uauxdot_zero)`, as this should already contain independent and dependent generalized speeds. However we can at least immediately say that all the terms this equation leaves are linearly with the normal `udots`. This means that they will all be substituted to zero. Therefore we can remove term 2 from the equation.

For term 3 we know it does not have any `udot_zero` terms what so ever, since the were already substituted [here](https://github.com/sympy/sympy/blob/4f48aad04cdb34e31e821ef9f6e9aa1af0ac637d/sympy/physics/mechanics/kane.py#L440). Therefore we can savely rewrite it to `nonMM`, dropping the `msubs`.

This leads to the result:
```
self._f_d = -(self._fr - dep_transform(nonMM)))
```

Conclusion: I'm quite certain that it does manage the auxilary speeds correctly, assuming nothing was already broken.

Reply:
Ok, I'm convinced. Thanks for carefully examining this.